samples/nrf9160/gnss/src/main.c

```diff
@@ -273,14 +279,22 @@ static int sample_init(void)
 	k_work_init(&agps_data_get_work, agps_data_get);
 
 	err = assistance_init(&agps_work_q);
-#endif
+#endif /* !CONFIG_GNSS_SAMPLE_ASSISTANCE_NONE */
 
 	return err;
 }
 
 static int gnss_init_and_start(void)
 {
-	/* Configure GNSS */
+#if defined(CONFIG_GNSS_SAMPLE_ASSISTANCE_NONE) || defined(CONFIG_GNSS_SAMPLE_LTE_ON_DEMAND)
+	/* Enable GNSS. */
+	if (lte_lc_func_mode_set(LTE_LC_FUNC_MODE_ACTIVATE_GNSS) != 0) {
+		LOG_ERR("Failed to enable GNSS");
```

Comment:
To me "Failed to change modem to GNSS mode" sounds like it would be related to the system mode, which is not the case here. The system mode is already configured by LTE LC so that GNSS support is enabled. This function call sends `AT+CFUN=31` to the modem, which "activates GNSS without changing LTE". The comment can be updated, but it should be such that it would not be confused with setting the system mode instead. What about "Failed to activate GNSS functional mode"?

Reply:
Yep, that's fine.